tests/unit/Imports/ProductImportTest.php

```diff
@@ -0,0 +1,134 @@
+<?php
+
+namespace StoreKeeper\WooCommerce\B2C\UnitTest\Imports;
+
+use Mockery\MockInterface;
+use StoreKeeper\WooCommerce\B2C\Commands\ProcessAllTasks;
+use StoreKeeper\WooCommerce\B2C\Imports\AbstractProductImport;
+use StoreKeeper\WooCommerce\B2C\Options\StoreKeeperOptions;
+use StoreKeeper\WooCommerce\B2C\Tools\StoreKeeperApi;
+use StoreKeeper\WooCommerce\B2C\UnitTest\Commands\AbstractTest;
+use Throwable;
+
+class ProductImportTest extends AbstractTest
+{
+    public const MEDIA_DATADUMP_DIRECTORY = 'imports/products/media';
+
+    public const CREATE_DATADUMP_SUCCESS_HOOK = 'imports/hook.events.createProduct.success.json';
+    public const CREATE_DATADUMP_FAILED_HOOK = 'imports/hook.events.createProduct.failed.json';
+
+    public const CREATE_DATADUMP_DIRECTORY = 'imports/products/createProduct';
+    public const CREATE_DATADUMP_SUCCESS_PRODUCT = 'moduleFunction.ShopModule::naturalSearchShopFlatProductForHooks.success.json';
+    public const CREATE_DATADUMP_FAIL_PRODUCT = 'moduleFunction.ShopModule::naturalSearchShopFlatProductForHooks.failed.json';
+
+    public function dataProviderTestImportWithStatusReporting(): array
+    {
+        $tests = [];
+
+        $tests['successful processing'] = [
+            self::CREATE_DATADUMP_SUCCESS_HOOK,
+            1,
+            1,
+            AbstractProductImport::SYNC_STATUS_SUCCESS,
+            8,
+        ];
+
+        $tests['failed processing'] = [
+            self::CREATE_DATADUMP_FAILED_HOOK,
+            0,
+            1,
+            AbstractProductImport::SYNC_STATUS_FAILED,
+            9,
+        ];
+
+        return $tests;
+    }
+
+    /**
+     * @dataProvider dataProviderTestImportWithStatusReporting
+     *
+     * @throws Throwable
+     */
+    public function testImportWithStatusReporting(
+        string $dumpHookFile,
+        int $expectedProductCount,
+        int $expectedStatusCallCount,
+        string $expectedStatus,
+        int $expectedShopProductId
+    ): void {
+        $setShopProductObjectSyncStatusForHookCallCount = 0;
+
+        StoreKeeperApi::$mockAdapter
+            ->withModule(
+                'ShopModule',
+                function (MockInterface $module) use (
+                    $expectedStatus,
+                    $expectedShopProductId,
+                    &$setShopProductObjectSyncStatusForHookCallCount
+                ) {
+                    $module->allows('setShopProductObjectSyncStatusForHook')
+                        ->andReturnUsing(function ($got) use (
+                            $expectedStatus,
+                            $expectedShopProductId,
+                            &$setShopProductObjectSyncStatusForHookCallCount
+                        ) {
+                            $request = $got[0];
+                            $this->assertEquals($expectedStatus, $request['status'], 'Reported status is wrong');
```

Comment:
Just a comment.

Not wrong but a bit confusing for devs used to messages that equate with the assertion's result, but I think I get why the message is negative but the assertion is positive (In case the assertion failed).

In the backend, we use messages as a way to explain what is the assertion for, in this project if this is the way it has been since we can keep it. But I think this should also be considered in our coding convention regarding whether to use positive or negative messages.

Reply:
Yes, maybe based on the project but I started the unit tests like this as I have also read a thread before stating
`The message is meant to be used in case of a test failure`. Maybe we can clarify with @ikulis as well.